homeassistant/components/decora_wifi/light.py

```diff
@@ -2,90 +2,40 @@
-from decora_wifi import DecoraWiFiSession
-from decora_wifi.models.person import Person
-from decora_wifi.models.residence import Residence
-from decora_wifi.models.residential_account import ResidentialAccount
-import voluptuous as vol
-
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS,
     ATTR_TRANSITION,
-    PLATFORM_SCHEMA,
     SUPPORT_BRIGHTNESS,
     SUPPORT_TRANSITION,
     LightEntity,
 )
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, EVENT_HOMEASSISTANT_STOP
-import homeassistant.helpers.config_validation as cv
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .common import DecoraWifiEntity, DecoraWifiPlatform
+from .const import DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-# Validation of the user's configuration
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Required(CONF_USERNAME): cv.string, vol.Required(CONF_PASSWORD): cv.string}
-)
 
-NOTIFICATION_ID = "leviton_notification"
-NOTIFICATION_TITLE = "myLeviton Decora Setup"
-
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Decora WiFi platform."""
-
-    email = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    session = DecoraWiFiSession()
-
-    try:
-        success = session.login(email, password)
-
-        # If login failed, notify user.
-        if success is None:
-            msg = "Failed to log into myLeviton Services. Check credentials."
-            _LOGGER.error(msg)
-            hass.components.persistent_notification.create(
-                msg, title=NOTIFICATION_TITLE, notification_id=NOTIFICATION_ID
-            )
-            return False
-
-        # Gather all the available devices...
-        perms = session.user.get_residential_permissions()
-        all_switches = []
-        for permission in perms:
-            if permission.residentialAccountId is not None:
-                acct = ResidentialAccount(session, permission.residentialAccountId)
-                for residence in acct.get_residences():
-                    for switch in residence.get_iot_switches():
-                        all_switches.append(switch)
-            elif permission.residenceId is not None:
-                residence = Residence(session, permission.residenceId)
-                for switch in residence.get_iot_switches():
-                    all_switches.append(switch)
-
-        add_entities(DecoraWifiLight(sw) for sw in all_switches)
-    except ValueError:
-        _LOGGER.error("Failed to communicate with myLeviton Service")
-
-    # Listen for the stop event and log out.
-    def logout(event):
-        """Log out..."""
-        try:
-            if session is not None:
-                Person.logout(session)
-        except ValueError:
-            _LOGGER.error("Failed to log out of myLeviton Service")
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
```

Comment:
```suggestion
    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
```

Reply:
Unresolved because I'm reverting this. It caused the following runtime error in my testing enviornment:
```
2021-09-19 00:13:16 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform decora_wifi.light: Platform not found (cannot import name 'AddEntitiesCallback' from 'homeassistant.helpers.entity_platform' (/usr/src/homeassistant/homeassistant/helpers/entity_platform.py)).
```

Reply:
After further reading: Could also be a namespace conflict?  I could try 
```
from homeassistant.helpers import entity_platform
.
.
.
async def async_setup_entry(
    hass: HomeAssistant, entry: ConfigEntry, async_add_entitiesentity_platform.AddEntitiesCallback
)
```
and see if that works.

Reply:
https://developers.home-assistant.io/docs/development_catching_up/#:~:text=If%20it's%20taking%20a%20while,of%20your%20commits%20on%20top.

Reply:
Ah. Given that it was added that recently I bet I need to update my hass docker containers, too.

Reply:
Confirmed my suspicions above; the docker image for my testing environment was stale. Pushing the patch now.